### PR TITLE
Add automatic conversions for unicode to std::string in Python 2

### LINKED ISSUE #23959: Workbench: setComment does not work in script editor
### Steps to reproduce the behavior

1. Start **Workbench**

2. Run the following code in **Editor**:
```python
ws = CreateSampleWorkspace()
ws.setComment('MyWorkspace')
```
3. Run the same code in **IPython**

### Expected behavior

Code runs successfully

### Actual behavior

The code run from the Editor produces the following error message:
```
CreateSampleWorkspace-[Notice] CreateSampleWorkspace started
CreateSampleWorkspace-[Notice] CreateSampleWorkspace successful, Duration 0.01 seconds
ArgumentError: Python argument types in
    Workspace.setComment(Workspace2D, unicode)
did not match C++ signature:
    setComment(Mantid::API::Workspace {lvalue} self, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > comment)
  File "", line 13, in <module>
```

No error is produced for the same code run in IPython.

### Platforms affected

Ubuntu 16.04, Python 2.7

The error does not happen on arch linux, Python 3.7

### LINKED ISSUE #23964: Workbench: Run->getLogData() does not work in script editor
Issue is very similar to #23959, just for different component.

### Steps to reproduce the behavior

1. Start **Workbench**

2. Run the following code in **Editor**:
```python
ws = CreateSampleWorkspace()
print(ws.getRun().getLogData('run_title').value)
```
3. Run the same code in **IPython**

### Expected behavior

Code runs successfully

### Actual behavior

The code run from the Editor produces the following error message:
```
CreateSampleWorkspace successful, Duration 0.01 seconds
Unlocking ws
ArgumentError: Python argument types in
    Run.getLogData(Run, unicode)
did not match C++ signature:
    getLogData(Mantid::API::Run {lvalue} self)
    getLogData(Mantid::API::Run {lvalue} self, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > name)
  File "", line 13, in <module>
```

No error is produced for the same code run in IPython.

### Platforms affected

Ubuntu 16.04, Python 2.7

The error does not happen on arch linux, Python 3.7

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Registers an additional converter for Python 2 that allows a `std::string` to be extracted from a `unicode` object automatically by the `boost::python` framework. The `unicode` object is encoded to a bytes object using "utf-8" and then extracted to `std::string`. Any encoding errors will be raised to Python as normal.

This also removes the need for the manual proxy methods that were doing this behaviour at each separate extraction point.

**Report to:** [nobody]. 

**To test:**

Verify that the examples in the referenced issues work in the workbench script editor.

Check the unit tests cover the appropriate methods.

Fixes #23959 
Fixes #23964 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by DTasev** [comment]
This fixes it for ASCII characters, however, if an actual unicode character is passed it raises an exception and freezes the script editor.

Example:
```python
ws = CreateSampleWorkspace()
print(ws.getRun().getLogData('run_title é').value)
```

This behaviour is different from MantidPlot where it raises 
```
RuntimeError: Unknown property search object run_titleé
  at line 2 in 'New script'
```

**Comment by martyngigg** [comment]
I think I have fixed the issue with non-ascii characters in the script editor for the workbench

**Comment by peterfpeterson** [comment]
Much of mantid is using `std::string` which I think is utf-8. Wouldn't we need to fundamentally change to `QString` everywhere to properly support unicode?

File: Framework/PythonInterface/mantid/kernel/src/Exports/Logger.cpp [hunk 1]
```diff
@@ -28,33 +27,33 @@ boost::shared_ptr<Logger> getLogger(const std::string &name) {
...
-boost::shared_ptr<Logger> create(const boost::python::object &name) {
-  return boost::make_shared<Logger>(Converters::pyObjToStr(name));
+boost::shared_ptr<Logger> create(const std::string &name) {
+  return boost::make_shared<Logger>(name);
 }
 
-void fatal(Logger *self, const boost::python::object &message) {
-  self->fatal(Converters::pyObjToStr(message));
-}
+// void fatal(Logger *self, const boost::python::object &message) {
```
Comment (peterfpeterson) [inline Framework/PythonInterface/mantid/kernel/src/Exports/Logger.cpp]: Remove commented out code from this file

**Comment by martyngigg** [comment]
For proper support yes. But we can at least have strings decoded with escape sequences that will fit into a `std::string`. It's the best we can do without changing to `std::wstring` absolutely everywhere.

**Comment by martyngigg** [comment]
For proper support yes. But we can at least have strings decoded with escape sequences that will fit into a `std::string`. It's the best we can do without changing to `std::wstring` absolutely everywhere.

**Comment by martyngigg** [comment]
At the moment I was concerned by allowing `from __future__ import unicode_literals` to work within Python 2 without having to hand convert every export function to essentially do what this would do anyway.

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
This is a much more succinct/universal way to handle unicode <-> utf8.

Tiny code cleanup left to do.

File: Framework/PythonInterface/mantid/kernel/src/Exports/Logger.cpp (hunk 1 above)
Comment (martyngigg) [reply Framework/PythonInterface/mantid/kernel/src/Exports/Logger.cpp]: Thanks!

**Review by peterfpeterson — APPROVED** [review]

**Review by dtasev — APPROVED** [review]
The changes fix the errors from the referenced issues. Additionally, the script editor will not freeze when unicode chars are pasted in it. LGTM :shipit:

**Pull request merged**